src/plugins/chart_expressions/expression_tagcloud/public/components/tagcloud_component.tsx

```diff
@@ -147,17 +143,19 @@ export const TagCloudChart = ({
           : getColor(palettesRegistry, palette, tag, values, syncColors) || 'rgba(0,0,0,0)',
       };
     });
+    // eslint-disable-next-line react-hooks/exhaustive-deps
   }, [
     bucket,
-    bucketFormatter,
-    metric,
-    palette,
-    palettesRegistry,
-    syncColors,
     visData.columns,
     visData.rows,
-    colorMapping,
+    metric,
+    // palettes, // somehow causes infinite update of RO
```

Comment:
should we double check this and understand why this happens? could it be due to the fact that the returned palettes are always a new object?

Reply:
I'm not sure what was the issue but removing the override and adding the `palettes` dep back, it seems to work fine. I even tried putting back the palette new object changes and had no issues. 🤷🏼 

Removed this in 600912fbbde5baafa72e444394749c426538ab86.